tests/ui/needless_lifetimes.rs

```diff
@@ -369,4 +369,47 @@ mod issue6159 {
     }
 }
 
+mod issue7296 {
+    use std::rc::Rc;
+    use std::sync::Arc;
+
+    struct Foo;
+    impl Foo {
+        fn implicit<'a>(&'a self) -> &'a () {
+            &()
+        }
+        fn implicit_mut<'a>(&'a mut self) -> &'a () {
+            &()
+        }
+
+        fn explicit<'a>(self: &'a Arc<Self>) -> &'a () {
+            &()
+        }
+        fn explicit_mut<'a>(self: &'a mut Rc<Self>) -> &'a () {
```

Comment:
Side note: In issue https://github.com/rust-lang/rust/issues/69064 it was mentioned that elision lifetimes work for `Rc<Self>`. I think it's not worth to specifically check for `Rc<Self>` as that would add extra complexity with little benefit. It's better to keep this consistent and wait for the issue to be fixed :upside_down_face:

Reply:
By that they meant it doesn't require `#![feature(arbitrary_self_types)]` to use